base/loggers/Loggers.cpp

```diff
@@ -191,6 +192,25 @@ void Loggers::buildLoggers(Poco::Util::AbstractConfiguration & config, Poco::Log
     }
 }
 
+void Loggers::updateLoggerLevel(Poco::Util::AbstractConfiguration & config, Poco::Logger & logger)
+{
+  int level = Poco::Logger::parseLevel(config.getString("logger.level", "trace"));
+  if (level == logger_level)
+    return;
+
+  logger_level = level;
+  logger.setLevel(level);
+  logger.root().setLevel(level);
+
+  // Todo: It would be better to set level in Logger class inside by introducing
+  //       a new API which will only acquire the mutex and walk through the map
+  //       object once.
+  std::vector<std::string> names;
+  logger.root().names(names);
+  for (const auto & name : names)
+    logger.root().get(name).setLevel(level);
+}
```

Comment:
What about the levels for particular loggers?

Here: https://github.com/ClickHouse/ClickHouse/pull/18401/files#diff-43a52ce116732798748d49ce68e6e20f9cd1c47027327c635cefc7c2e870c69dR169

Probably you can move this code to this function, and then call this function from `buildLoggers`.

Reply:
Sorry, I missed the particular loggers since we never used it :(

The 'updateLoggerLevel' and 'buildLoggers' should be separated because 'buildLoggers' is a little **heavy**.  At the same, record every particular logger is urgly. I have no better idea except updating every particular logger before updating 'logger.level'.  _What's your opinion?_